src/ucp/api/ucp.h

```diff
@@ -4636,6 +4637,11 @@ typedef struct ucp_ep_attr {
      * UCS_ERR_NOT_CONNECTED will be returned.
      */
     struct sockaddr_storage remote_sockaddr;
+
+    /**
+     * Pointer to struct containing transport and device names used by this endpoint.
+     */
+    ucp_transports_t *transports;
```

Comment:
since ucp_transports_t structure is not extendable (it does not have field_mask), this should be a value and not a pointer

Reply:
If I make this a value instead of a pointer then I have to move my two new typedefs from ucp_defs.h to this header, and this structure becomes a variable size structure since the number of transport/device name pairs now contained in this structure will vary. Is that acceptable?

Reply:
1. it's ok to move (though ucp.h includes ucp_def.h)
2. 'entries' is a pointer so i would not expect the structure itself to grow

Reply:
Agree, I think it should be value